certbot/updater.py

```diff
@@ -0,0 +1,67 @@
+"""Updaters run at renewal"""
+import logging
+
+from certbot import errors
+from certbot import interfaces
+
+from certbot.plugins import selection as plug_sel
+
+logger = logging.getLogger(__name__)
+
+def run_generic_updaters(config, plugins, lineage):
+    """Run updaters that the plugin supports
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param plugins: List of plugins
+    :type plugins: `list` of `str`
+
+    :param lineage: Certificate lineage object
+    :type lineage: storage.RenewableCert
+
+    :returns: `None`
+    :rtype: None
+    """
+    try:
+        # installers are used in auth mode to determine domain names
+        installer, _ = plug_sel.choose_configurator_plugins(config, plugins, "certonly")
+    except errors.PluginSelectionError as e:
+        logger.info("Could not choose appropriate plugin: %s", e)
+        raise
```

Comment:
Hm. Maybe we don't want to `raise` here -- if a single lineage is broken, we should still be trying to update the others, no?

Reply:
This is handled by the pre-existing try-except block in `renewal.handle_renewal_request` which calls this method: https://github.com/certbot/certbot/blob/plugin_renew_updates/certbot/renewal.py#L434-L440

However you are right that we probably shouldn't raise this anyway as the error message displayed for the user by the block above might confusing and is definitely inaccurate. It tells that there was an error with the renewal, which isn't the case in this situation.

I removed the `raise` and instead changed the logger level to `warning` and made the log message a bit more informative.